File: BaselineProject/BaselineGroup7/BaselineGroup7/Book.cpp
//
//  Book.cpp
//  BaselineGroup7
//
//  Created by ubicomp7 on 8/27/15.
//  Copyright (c) 2015 ubicomp7. All rights reserved.
//

#include "Book.h"

void Book::setQuantity(uint32_t quantity)
{
    this->quantity = quantity;
}

std::string Book::getName()
{
    return this->bookName;
}

std::string Book::getISBN()
{
    return this->isbn;
}

File: Exercise2Group7/Exercise2Group7/Exercise2Group7/Book.m
//
//  Book.m
//  Exercise2Group7
//
//  Created by Jared Jones on 9/10/15.
//  Copyright © 2015 Jared Jones. All rights reserved.
//

#import "Book.h"

@implementation Book

- (instancetype) init{
    NSLog(@"Must instantiate using initWithStuff:");
    return nil;
}

- (instancetype) initWithStuff: (NSString *)isbn withTitle: (NSString *)title withAuthors: (NSString *)authors withCategory: (NSNumber *)category
{
    if ( self = [super init]){
        _isbn = isbn;
        _title = title;
        _authors = authors;
        _category = category;
        return self;
    }
    return nil;
}

+ (NSString *)getCategoryGivenID: (NSUInteger)num
{
    if (num == 0)
        return @"Programming";
    if (num == 1)
        return @"Science";
    if (num == 2)
        return @"Math";
    if (num == 3)
        return @"Fiction";

    return @"BAD BOY";
}

@end
